sources/rest_api/README.md

```diff
@@ -0,0 +1,222 @@
+
+And now the `pokemon` one. This is actually a child endpoint of the `pokemon_list`: for each pokemon we want to get further details. So we need to make this resource a bit more smart, the endpoint `path` needs to be explicit, and we have to specify how the value of `name` will be resolved from another resource; this is actually telling the generic source that `pokemon` needs to be queried for each pokemon in `pokemon_list`.
+
+## Anatomy of the config object
+
+> **_TIP:_**  Import `RESTAPIConfig` from the `rest_api` module to have convenient tips.
+
+The config object passed to the REST API Generic Source has three main elements:
+
+```python
+my_config: RESTAPIConfig = {
+    "client": {
+        ...
+    },
+    "resource_defaults": {
+        ...
+    },
+    "resources": {
+        ...
+    ,
+}
+```
+
+
+
+`client` contains the configuration to connect to the APIs endpoints (e.g. base URL, authentication method, default behaviour for the paginator, and more).
+
+`resource_defaults` contains the default values to configure the dlt resources returned by this source.
+
+`resources` object contains the configuration for each resource. 
+
+The configuration with smallers scope will overwrite the one with the wider one:
+
+    Resource Configuration > Resource Defaults Configuration > Client Configuration
+
+## Reference
+
+### `client`
+
+#### `auth` [optional]
+Use the auth property to pass a token or a `HTTPBasicAuth` object for more complex authentication methods. Here some practical examples:
+
+1. Simple token (read from the `.dlt/secrets.toml` file):
+```python
+my_api_config: RESTAPIConfig = {
+    "client": {
+        "base_url": "https://my_api.com/api/v1/",
+        "auth": {
+            "token": dlt.secrets["sources.my_api.access_token"],
+        },
+    },
+    ...
+}
+```
+
+2. 
+```python
+from requests.auth import HTTPBasicAuth
+
+basic_auth = HTTPBasicAuth(dlt.secrets["sources.my_api.access_token"], "")
+
+my_api_config: RESTAPIConfig = {
+    "client": {
+        "base_url": "https://my_api.com/api/v1/",
+        "auth": basic_auth,
+    },
+    ...
+}
+```
+
+#### `base_url`
+The base URL that will be prepended to the endpoints specified in the `resources` objects. Example
+
+```python
+    "base_url": "https://my_api.com/api/v1/",
+```
+
+#### `paginator` [optional]
+The paginator property specify the default paginator to be used for the endpoint responses.
+
+//TODO describe paginators, the possible strings, and the disctionary to be used 
+
+Possible paginators are:
+- BasePaginator - 
+- HeaderLinkPaginator - String alias `"header_links"` -

```

Comment:
I like the idea of having both classes and string aliases in a same place so it looks like a good place to use a markdown table.